common-rest/src/main/java/org/uniprot/api/rest/output/header/HttpCommonHeaderConfig.java

```diff
@@ -116,7 +119,9 @@ void handleGatewayCaching(HttpServletRequest request, HttpServletResponse respon
                 break;
             }
         }
-        if (requiresCachingHeaders) {
+        if (requiresCachingHeaders
+                && (HttpServletResponse.SC_OK == response.getStatus()
+                        || HttpServletResponse.SC_SEE_OTHER == response.getStatus())) {
             // request gateway caching
```

Comment:
Please filter specific codes which don't need cache

Reply:
I updated to < 400, as we discussed in our chat